### DISCUSSION TITLE
How to handle uncertainty in future Covariates?

### DISCUSSION THREAD

**Discussion by BenediktPrusas** [autogluon/autogluon#3891] [Q&A]
Hello everyone,

amazing library so far! But right now, I am unsure how to handle uncertainty in future covariates. I want to use historic weather forecasts as future covariates, so they are evolving over time. If I only use the actual weather data, the learned probability distribution will not reflect the uncertainty in of the weather forecast and be overconfident. 

So far, I thought I could create one timeseries for every training example, where each time series is only long enough to make exactly one prediction. But this isn't exactly elegant. 

Similarly, I have the situation that the reported values of my time series might get corrected in the first days (e.g. failures of measurements equipment etc.), also here the model could learn to be overconfident in the reported past target values. Here I would need to add a series for every correction that occurred and make it small enough that it is exactly only one training sample long.

Are there any better ways to approach this? Cutom Datasets or DataLoaders, I would be also fine to moving to GluonTS or only using Deep Learning Models.

Thanks in advance!

**Comment by shchur** [comment]
Hi, this is a really good question to which I, unfortunately, don't have a good answer. Most likely, the answer heavily depends on your application / dataset / accuracy of your forecast for the unknown covariates. I would recommend evaluating 3 versions of the `TimeSeriesPredictor` trained with different data:
1.  passing the predicted future values of the covariates as `known_covariates` (i.e., provide the names of uncertain covariates as `known_covariates_names` when creating predictor + provide predicted future values as `known_covariates` when calling `predict`).
2. only using the `past_covariates` (i.e., don't provide the names of the uncertain covariates as `known_covariates_names`).
3. completely removing all uncertain covariates from the dataset

I would then compare the performance of the 3 versions on some held-out data and select the one that works best for your application.
